test/test_unistd.rs

```diff
@@ -624,6 +624,27 @@ fn test_sysconf_unsupported() {
     assert!(open_max.expect("sysconf failed").is_none())
 }
 
+
+#[cfg(target_os = "linux")]
```

Comment:
libc also implements these functions on Android.  But they're actually implemented on some of the BSDs, too.  Best would be to make a PR to libc that adds bindings everywhere they're implemented, then broaden your `cfg` restriction in Nix appropriately.